### PR TITLE
Fix typos, add some tests

### PR DESCRIPTION
**Pull request by jgsogo**
I think that with this PR yours will be completed, and then we can merge it into mine. Something else to test?

Thanks!

### CODE REVIEW

File: conans/test/util/tools_test.py [hunk 1]
```diff
@@ -1115,6 +1115,28 @@ def test_repo_root(self):
...
+
+        git.run('commit -m "commit"')
+        self.assertTrue(git.is_pristine())
+
+    def test_is_local_repository(self):
+        root_path, _ = create_local_git_repo({"myfile": "anything"})
+
+        git = Git(root_path)
+        self.assertTrue(git.is_local_repository())
+        # TODO: Check that with remote one it is working too
```
Comment (jgsogo) [inline conans/test/util/tools_test.py]: I would want to check this, but I don't know how to mock a remote git repo :/

File: conans/test/util/tools_test.py [hunk 2]
```diff
@@ -1442,6 +1464,67 @@ def test_repo_root(self):
         self.assertFalse(svn2.folder == tmp_folder)
         self.assertEqual(os.path.realpath(tmp_folder).lower(), svn2.get_repo_root().lower())
 
+    def test_is_local_repository(self):
+        svn = SVN(folder=self.gimme_tmp())
+        svn.clone(url=self.repo_url)
+        self.assertTrue(svn.is_local_repository())
+
+        # TODO: Test not local repository
```
Comment (jgsogo) [inline conans/test/util/tools_test.py]: Same as above, but for SVN

File: conans/test/util/tools_test.py (hunk 1 above)
Comment (climblinne) [reply conans/test/util/tools_test.py]: May be we should ask the conan team, if they have some test repositories, which we can use for testing.

File: conans/test/util/tools_test.py (hunk 2 above)
Comment (climblinne) [reply conans/test/util/tools_test.py]: Same...

Comment (climblinne) [reply conans/test/util/tools_test.py]: I think github supports also SVN on a Git repository. May be just create a dummy repo or take the main conan repo for testing

Comment (climblinne) [reply conans/test/util/tools_test.py]: Or can we use one of the offical examples repos?

**Pull request merged**